keras/applications/nasnet.py

```diff
@@ -210,24 +210,18 @@ def NASNet(input_shape=None,
     channel_dim = 1 if K.image_data_format() == 'channels_first' else -1
     filters = penultimate_filters // 24
 
-    if not skip_reduction:
```

Comment:
I recommend not removing these if blocks. Without these checks, CIFAR models will be reduced from 32x32 to 4x4 prior to the first normal_a cell.
Instead : 
1) Rename `skip_reduction` to `skip_reduction_input_layer` to match TF
2) Set the `if not skip_reduction` to `if skip_reduction_input_layer` to allow only the Large model to use them. 
3) Set the NASNet Mobile version to **not** use `skip_reduction_input_layer`, according to the TF code.